cli/src/commands/create/dataset.rs

```diff
@@ -1,15 +1,30 @@
 use failchain::ResultExt;
+use lazy_static::lazy_static;
 use log::info;
+use regex::Regex;
 use reinfer_client::{
     resources::comment::EntityKind, Client, DatasetFullName, NewDataset, SourceIdentifier,
 };
 use structopt::StructOpt;
 
 use crate::errors::{ErrorKind, Result};
 
+lazy_static! {
+    static ref RX_FULL_NAME: Regex =
+        Regex::new("^[A-Za-z0-9-_.]{1,1024}/[A-Za-z0-9-_]{1,256}").unwrap();
```

Comment:
I thought we would want to limit this to a shorter name overall due to API constraints? cc: @annesylvie and her recent api validation work.

I'm also happy to leave this more permissible and just return the API error if the name is too long.

Reply:
Currently, with this regex you can get an error like:
```bash
$ cargo run -- create dataset org1/some-incredibly-long-dataset-name-that-will-cause-a-500-error-in-the-api-because-its-over-127-chars-long-no-really-now-its-over-the-limit
E Operation to create a dataset has failed.
E  |- API request failed with 422 Unprocessable Entity: Dataset name is too long; the maximum length of the string org1/<dataset_name> is of 127 characters
```
which I think is absolutely fine

Reply:
Maybe I combined the wrong regex? I pulled this from the API code, happy to update it to something even more restrictive if that's the way to go.

Reply:
I did indeed add a new constraint on the length of the string `<org>/<dataset_name>` that is accepted by the api; I guess this was merged after you started working on this!

The regex itself is unchanged; we check for the length of the string separately. Up to you if you also want to implement this behaviour @scallaway. It can also be another PR if you prefer.

Reply:
I think in that case it's fine to leave the regex as is. We expect to surface API errors directly to users in a lot of cases, the validation in the CLI should be expected to be light (similar to frontend validation)